src/converter.coffee

```diff
@@ -59,36 +61,100 @@ class Converter
     return output
 
   #@private
-  getStateAndMessageFromAmandaResult: ({pathArray, amandaResult, lowerCasedKeys}) ->
-    resultsCount = amandaResult.length or 0
+  # aggregate all errors for given path and mark them with state missing, added, changed etc..
+  getStateAndMessageFromAmandaResult: ({pathArray, lowerCasedKeys}) ->
+    resultsCount = @gavelResult.rawData.length or 0
     state = 0
     message = ''
 
-    if resultsCount
-      if lowerCasedKeys
-        transformKeys = (s) ->
-          s.toLowerCase()
-      else
-        transformKeys = (s) ->
-          s
+    if lowerCasedKeys
+      transformKeys = (s) ->
+        s.toLowerCase()
+    else
+      transformKeys = (s) ->
+        s
 
-      for i in [0..resultsCount-1]
-        if @areArraysIdentical pathArray.map(transformKeys), amandaResult[i]?['property'] or []
 
-          if amandaResult[i]['validator'] == 'required'
+    # get all pointers in expected data
+    dataExpectedPointers = []
+    traverse(@dataExpected).forEach (nodeValue) ->
+      dataExpectedPointers.push jsonPointer.compile this.path.map(transformKeys)
+      return
+
+    dataRealPointers = []
+    # get all pointers in expected data
+    traverse(@dataReal).forEach (nodeValue) ->
+      dataRealPointers.push jsonPointer.compile this.path.map(transformKeys)
+      return
+
+    if resultsCount
```

Comment:
can the `traverse` above (to obtain `dataRealPointers`) be moved into this `if` statement?